packages/web/src/components/result/ResultCard.js

```diff
@@ -41,6 +41,18 @@ class ResultCard extends Component {
 		this.internalComponent = `${props.componentId}__internal`;
 	}
 
+	componentWillMount() {
```

Comment:
It'd be best if we do this evaluation in the constructor and kill the unnecessary noise.

Reply:
@metagrover I made the `defaultPage` undefinable so that I could use it to detect whether `defaultPage` has been set from the url query value and, if not set, then ignore it and prefer the use of the `currentPage` prop on component initialization. The problem is that when the `defaultPage` can be set with a default value of 0, I cannot determine whether the reason for the value is because the url query value is 1 or whether the `defaultPage` defaulted to 0.  I need a way on initialization to determine whether the `defaultPage` should be ignored or used.

To allow the defaultPage prop to default to 0, I believe I can change the code to make the `defaultPage` valid values start at 1 and interpret 0 as ignore. To accomplish this, I can change the `defaultPage` portion of code in `mapStateToProps` to remove the subtract 1 from the selectedValue and set the default value to 0.  Here is what the code can be:

```
defaultPage: (
    state.selectedValues[`${props.componentId}-page`]
	&& state.selectedValues[`${props.componentId}-page`].value
) || 0,
```

Reply:
I agree, maybe we can do `-1` instead of `undefined`.